src/matcher_combinators/matchers.clj

```diff
@@ -1,31 +1,39 @@
 (ns matcher-combinators.matchers
   (:require [matcher-combinators.core :as core]))
 
-(defn equals-value
+(defn equals
   "Matcher that will match when the given value is exactly the same as the
   `expected`."
   [expected]
-  (core/->Value expected))
+  (cond
+    (sequential? expected) (core/->EqualsSeq expected)
+    (set? expected)        (core/->EqualsSet expected false)
+    (map? expected)        (core/->EqualsMap expected)
+    :else                  (core/->Value expected)))
+
+(defn equals-set
+  "Matches a set in the way `(equals some-set)` would, but accepts sequences as
+  the expected matcher argument, allowing one to use matchers with the same
+  submatcher appearing more than once."
+  [expected]
+  (core/->EqualsSet expected true))
 
-(defn contains-map
+(defn embeds
   "Matcher that will match when the map contains some of the same key/values as
   the `expected` map."
   [expected]
-  (core/->ContainsMap expected))
-
-(defn equals-map
-  "Matcher that will match when:
-    1. the keys of the `expected` map are equal to the given map's keys
-    2. the value matchers of `expected` map matches the given map's values"
-  [expected]
-  (core/->EqualsMap expected))
-
-(defn equals-seq
-  "Matcher that will match when the `expected` list's matchers match the given list.
-
-  Similar to Midje's `(just expected)`"
+  (cond
+    (sequential? expected) (core/->EmbedsSeq expected)
+    (set? expected)        (core/->EmbedsSet expected false)
+    (map? expected)        (core/->EmbedsMap expected)
+    :else                  (core/->InvalidType expected "embeds" "seq, set, map")))
+
+(defn embeds-set
+  "Matches a set in the way `(embeds some-set)` would, but accepts sequences
+  as the expected matcher argument, allowing one to use matchers with the same
+  submatcher appearing more than once."
```

Comment:
Do you have real use cases for `embeds-set`? At first sight, it seems to be more specific and a bit harder to understand than the rest of the matchers. 

BTW, is it `embeds-set` or `set-embeds`?